frappe/public/js/frappe/social/pages/Profile.vue

```diff
@@ -20,35 +40,51 @@ export default {
 	data() {
 		return {
 			user_id: '',
-			my_posts: []
+			posts: [],
+			current_list: 'posts'
+			
 		}
 	},
 	created() {
 		this.user_id = frappe.get_route()[2];
 		this.set_user_posts();
-		frappe.route.on('change', () => {
-			this.user_id = frappe.get_route()[2];
-			this.set_user_posts()
-		});
+	},
+	computed: {
+		pinned_posts() {
+		 	return this.posts.filter((post) => post.is_pinned&&post.owner===this.user_id)
+		},
+		get_posts(){
+			return this.posts.filter((post) => post.owner === this.user_id)
+		}, 
+		get_liked_posts(){
+			return this.posts.filter((post) => post.liked_by.includes(this.user_id))
+		}
+		
 	},
 	methods: {
 		set_user_posts() {
 			frappe.db.get_list('Post', {
-				filters: {
-					owner: this.user_id,
-				},
 				fields: ['name', 'content', 'owner', 'creation', 'liked_by', 'is_pinned'],
 				order_by: 'creation desc',
 			}).then((posts) => {
-				this.my_posts = posts;
+				this.posts = posts;
```

Comment:
It's not good to get all the posts and then add the filter... 

You can instead write a custom whitelisted method on the server side to get different filtered records and you just have to make a single call from the client side.